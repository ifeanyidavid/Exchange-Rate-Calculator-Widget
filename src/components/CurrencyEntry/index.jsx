import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { fieldNames } from "../../constants/fields";

const SliderWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  min-height: 200px;
  max-height: 100%;
  padding: 16px;
  cursor: move;

  .balance {
    min-width: 100px;
    max-width: 100%;
  }

  .currency {
    font-size: 24px;
    font-weight: bold;
    text-transform: uppercase;
  }

  .currency-input {
    text-align: right;
    box-shadow: none;
    font-size: 18px;
    padding: 20px;
    width: 230px;
    outline: none !important;
    border: none;
  }

  .currency-input:focus {
    box-shadow: none;
  }
`;

function CurrencyEntry({
  name,
  onChange,
  value,
  balance,
  currency,
  symbol,
  showExchangeRate,
  destinationRate,
  ratesFetched
}) {
  const prefix = name === fieldNames.sourceAmount ? "-" : "+";

  const disabled = ratesFetched();
  return (
    <SliderWrapper>
      <div className="d-flex flex-column balance">
        <h2 className="currency">{currency}</h2>
        <small className="text-muted">
          You have {symbol} {parseInt(balance).toString()}
        </small>
      </div>
      <div className="d-flex flex-column justify-content-end">
        <input
          type="text"
          pattern="[0-9]*"
          placeholder="0"
          autoComplete="off"
          className="currency-input"
          name={name}
          value={value ? prefix + value : value}
          onChange={e => onChange(e)}
          disabled={disabled}
        />
        {showExchangeRate && (
          <small className="font-weight-bold text-right">
            {destinationRate}
          </small>
        )}
      </div>
    </SliderWrapper>
  );
}

export default CurrencyEntry;

CurrencyEntry.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  balance: PropTypes.number.isRequired,
  currency: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  showExchangeRate: PropTypes.bool.isRequired,
  destinationRate: PropTypes.string,
  ratesFetched: PropTypes.func
};
