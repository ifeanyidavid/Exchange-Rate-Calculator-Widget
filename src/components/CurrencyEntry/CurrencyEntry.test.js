import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import CurrencyEntry from ".";

configure({ adapter: new Adapter() });

describe("CurrencyEntry component", () => {
  it("Renders correctly", () => {
    const props = {
      name: "source",
      onChange: jest.fn(),
      value: "1.1",
      balance: 300.0,
      currency: "USD",
      symbol: "$",
      showExchangeRate: true,
      destinationRate: "1.12",
      ratesFetched: jest.fn()
    };

    // wrapper instance around rendered output
    const wrapper = shallow(<CurrencyEntry {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
