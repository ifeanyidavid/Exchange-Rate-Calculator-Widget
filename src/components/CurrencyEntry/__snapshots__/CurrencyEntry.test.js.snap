// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyEntry component Renders correctly 1`] = `
<styled.div>
  <div
    className="d-flex flex-column balance"
  >
    <h2
      className="currency"
    >
      USD
    </h2>
    <small
      className="text-muted"
    >
      You have 
      $
       
      300
    </small>
  </div>
  <div
    className="d-flex flex-column justify-content-end"
  >
    <input
      autoComplete="off"
      className="currency-input"
      name="source"
      onChange={[Function]}
      pattern="[0-9]*"
      placeholder="0"
      type="text"
      value="+1.1"
    />
    <small
      className="font-weight-bold text-right"
    >
      1.12
    </small>
  </div>
</styled.div>
`;
